File: Old Java Code/Client/org/hardgate/familytree/formatters/CensusOutputFormatter.java
/*
 * Created on 14-Jan-2005
 *
 */
package org.hardgate.familytree.formatters;

import java.io.PrintWriter;
import java.util.Collections;
import java.util.List;

import org.hardgate.familytree.comparators.CensusAgeComparator;
import org.hardgate.familytree.core.Individual;
import org.hardgate.familytree.registrations.CensusRegistration;
import org.hardgate.familytree.registrations.Registration;

/**
 * @author a-bisset
 *
 */
public class CensusOutputFormatter extends BaseOutputFormatter {

    public void printHeader(PrintWriter output) {
        output.print("FamilyGed,BestLocation,CensusName,Age,");
        output.print("Occupation,DateOfBirth,BirthLocation,Status");
        output.println();
    }

    public void printItem(Registration reg, PrintWriter output) {
        CensusRegistration c = (CensusRegistration) reg;
        List<Individual> members = c.getMembers();
        Collections.sort(members, new CensusAgeComparator());
        for (Individual i : members) {
            output.print("\""); output.print(c.getFamilyGed()); output.print("\",");
            output.print("\""); output.print(c.getRegistrationLocation()); output.print("\",");
            output.print("\""); output.print(i.getCensusName()); output.print("\",");
            output.print("\""); output.print(i.getAge(c.getRegistrationDate())); output.print("\",");
            output.print("\""); output.print(i.getOccupation()); output.print("\",");
            output.print("\""); output.print(i.getDateOfBirth()); output.print("\",");
            output.print("\""); output.print(i.getBirthLocation()); output.print("\",");
            output.print("\""); output.print(i.getStatus()); output.print("\"");
            output.println();
        }
    }
}

File: FTAnalyser/Formatters/CensusOutputFormatter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Xml;
using System.IO;

namespace FTAnalyser
{
    public class CensusOutputFormatter : BaseOutputFormatter {

        public override void printHeader(StreamWriter output)
        {
            output.Write("FamilyGed,BestLocation,CensusName,Age,");
            output.Write("Occupation,DateOfBirth,BirthLocation,Status");
            output.WriteLine();
        }

        public override void printItem(Registration reg, StreamWriter output)
        {
            CensusRegistration c = (CensusRegistration) reg;
            List<Individual> members = c.Members;
            members.Sort(new CensusAgeComparator());
            foreach (Individual i in members) {
                output.Write("\""); output.Write(c.FamilyGed); output.Write("\",");
                output.Write("\""); output.Write(c.RegistrationLocation); output.Write("\",");
                output.Write("\""); output.Write(i.CensusName); output.Write("\",");
                output.Write("\""); output.Write(i.getAge(c.RegistrationDate)); output.Write("\",");
                output.Write("\""); output.Write(i.Occupation); output.Write("\",");
                output.Write("\""); output.Write(i.DateOfBirth); output.Write("\",");
                output.Write("\""); output.Write(i.BirthLocation); output.Write("\",");
                output.Write("\""); output.Write(i.Status); output.Write("\"");
                output.WriteLine();
            }
        }
    }
}